tests/baselines/reference/user/clone.log

```diff
@@ -7,6 +7,7 @@ node_modules/clone/clone.js(176,14): error TS2532: Object is possibly 'undefined
 node_modules/clone/clone.js(186,16): error TS2403: Subsequent variable declarations must have the same type.  Variable 'i' must be of type 'string', but here has type 'number'.
 node_modules/clone/clone.js(186,23): error TS2365: Operator '<' cannot be applied to types 'string' and 'number'.
 node_modules/clone/clone.js(186,52): error TS2356: An arithmetic operand must be of type 'any', 'number', 'bigint' or an enum type.
+node_modules/clone/clone.js(255,35): error TS2774: This condition will always return true since this function is always defined. Did you mean to call it instead?
```

Comment:
This error is totally wrong - the pattern is
```js
if (typeof module === 'object' && module.exports) {
  module.exports = clone;
}
```
 - the inner `module.exports` assignment being treated as unconditional is making us error on the check to perform the assignment.
Minimally, we need to add an exception to the error where `module.exports` checks, specifically, aren't considered always-defined (since they're runtime module system checks).

Reply:
https://github.com/microsoft/TypeScript/issues/48166